lib/matplotlib/backends/backend_tkagg.py

```diff
@@ -529,21 +532,44 @@ def __init__(self, canvas, num, window):
         self.window.withdraw()
         self.set_window_title("Figure %d" % num)
         self.canvas = canvas
-        self._num =  num
-        if matplotlib.rcParams['toolbar']=='toolbar2':
-            self.toolbar = NavigationToolbar2TkAgg( canvas, self.window )
-        else:
-            self.toolbar = None
-        if self.toolbar is not None:
-            self.toolbar.update()
         self.canvas._tkcanvas.pack(side=Tk.TOP, fill=Tk.BOTH, expand=1)
+        self._num = num
+
+        self.toolmanager = self._get_toolmanager()
+        self.toolbar = self._get_toolbar()
+        self.statusbar = None
+
+        if matplotlib.rcParams['toolbar'] == 'toolmanager':
+            tools.add_tools_2_toolmanager(self.toolmanager)
+            tools.add_tools_2_container(self.toolbar)
```

Comment:
We had a discussion with @tacaswell and @OceanWolf  about this. It is just a helper function that adds the default `tools.default_tools` to the given `toolmanager` and the `tools.default_toolbar_tools` to the given `toolcontainer`.

Before they were called `ToolManager.add_tools` and `ToolContainerBase.add_tools` but that was prone to confusion because the singular `add_tool` already existed and is more important.
When those methods where inside the classes, this "awkwardness" didn't happen